indexer/src/main/java/au/org/aodn/esindexer/service/StacCollectionMapperService.java

```diff
@@ -431,30 +428,13 @@ List<LinkModel> mapLinks(MDMetadataType source) {
         }
 
         // Now add links for logos
-        Optional<Map<String, Object>> optAdditionalInfo = geoNetworkService.getRecordExtraInfo(this.mapUUID(source));
-        if(optAdditionalInfo.isPresent()) {
-            // We iterate logos link and add it to STAC
-            Map<String, Object> additionalInfo = optAdditionalInfo.get();
-            if(additionalInfo.containsKey(SUGGEST_LOGOS)) {
-                if(additionalInfo.get(SUGGEST_LOGOS) instanceof List) {
-                    ((List<?>) additionalInfo.get(SUGGEST_LOGOS))
-                            .stream()
-                            .map(p -> (p instanceof String) ? (String) p : null)
-                            .filter(Objects::nonNull)
-                            .filter(i -> urlUtils.checkUrlExists(i))
-                            .findFirst()        // We only pick the first reachable one
-                            .ifPresent(i -> {
-                                LinkModel linkModel = LinkModel.builder().build();
-                                linkModel.setHref(i);
-                                // Geonetwork always return png logo
-                                linkModel.setType("image/png");
-                                linkModel.setRel("icon");
-                                linkModel.setTitle("Suggest icon for dataset");
-                                results.add(linkModel);
-                            });
-                }
-            }
-        }
+        geoNetworkService.getLogo(this.mapUUID(source))
+                .ifPresent(results::add);
+
+        // Thumbnail link
+        geoNetworkService.getThumbnail(this.mapUUID(source))
+                .ifPresent(results::add);
```

Comment:
so there will not be a corresponding fields inside the record (document in the ES index store) e.g

if there is data, it looks like this:
```json
{
  "thumbnails": ....
}
```

if there is no data:
if there is data, it looks like this:
```json
{
 ...
}
```

should the structure of the record to be persistent regardless there is data or not? e.g, if there are no thumbnails:
```json
{
  "thumbnails": null
}
```

Reply:
Add one more test case to verify thumbnails: [] empty will not crash es-indexer